Wrappers/Python/cil/framework/data_container.py

```diff
@@ -0,0 +1,1522 @@
+# -*- coding: utf-8 -*-
+#  Copyright 2024 United Kingdom Research and Innovation
+#  Copyright 2024 The University of Manchester
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+# Joshua DM Hellier (University of Manchester) [refactorer]
+
+import copy
+import ctypes
+import logging
+import warnings
+from functools import reduce
+from numbers import Number
+
+import numpy
+
+from .label import image_labels, data_order, get_order_for_engine
+from .cilacc import cilacc
+from cil.utilities.multiprocessing import NUM_THREADS
+
+
+def message(cls, msg, *args):
+    msg = "{0}: " + msg
+    for i in range(len(args)):
+        msg += " {%d}" %(i+1)
+    args = list(args)
+    args.insert(0, cls.__name__ )
+
+    return msg.format(*args )
+
+
+class ImageGeometry(object):
```

Comment:
for backward-compat:


```suggestion
class ImageGeometry(object):

    @property
    def HORIZONTAL_X(self):
        warnings.warn("use ... instead", DeprecationWarning, stacklevel=2)
        return image_labels['HORIZONTAL_X']
```